### DISCUSSION TITLE
Database connection

### DISCUSSION THREAD

**Discussion by jaspreet1313** [Budibase/budibase#4543] [Help]
I am creating CRUD app in budibase .but in second application (CRUD app using MySQL) how I create connection between MySQL and CRUD app.

**Comment by mjashanks** [comment]
Hey @jaspreet1313 can you give some more detail? 

You can create a connection to MySQL by adding a new Data source, and choosing MySQL - then completing the connection details.

![image](https://user-images.githubusercontent.com/3524181/154445076-a77398bf-1770-498a-9bb3-7478e8487bb4.png)

**Comment by jaspreet1313** [comment]
On Thu, 17 Feb, 2022, 2:52 pm Michael Shanks, ***@***.***>
wrote:

> Hey @jaspreet1313 <https://github.com/jaspreet1313> can you give some
> more detail?
>
> You can create a connection to MySQL by adding a new Data source, and
> choosing MySQL - then completing the connection details.
>
What I have to fill in connection detail.please tell me .because it shows
error saving datasource.

**Reply by mjashanks** [reply]
Just to be clear, if you are using MySQL, you need to have setup your own MySQL database somewhere. At that point, you will know your own connection details.

**Reply by jaspreet1313** [reply]
After connection with mysql in budibase . when i fetching tables it shows 'error updating datasource schema'.
How to resolve it . please help me.

**Reply by gerhardmpl** [reply]
Can you share the result of `mysql -V` for the version of MySQL (assuming you are using Linux to host MySQL)?
Did you check if a firewall is active and is allowing `3306/tcp`?

**Reply by jaspreet1313** [reply]
The result of mysql -V is "mysql  Ver 15.1 Distrib 10.3.34-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2".
yes I checked
The firewall is active and allowing 3306/tcp.
'https://user-images.githubusercontent.com/79891162/154848819-52b3aaa3-bff8-478c-90ac-d1a00ce007f2.png' Showing in this link.

**Reply by gerhardmpl** [reply]
Mh, looks like you are using MariaDB instead of MySQL? I have the same issues with MariaDB 10.5.12 (see #4581) and switched to MySQL 8.0 (which works - kind of).  It is a steep learning curve with not that much documentation around.
